tempto-core/src/main/java/com/teradata/tempto/s3/S3Client.java

```diff
@@ -0,0 +1,263 @@
+
+        for (String object : getAllObjects(ensureTrailingSlashPresent(path))) {
+            List<String> objectParts = Splitter.on('/').splitToList(ensureTrailingSlashRemoved(object));
+            if (objectParts.size() == pathCount + 1) {
+                objects.add(object);
+            }
+        }
+
+        return objects;
+    }
+
+    public boolean exists(String path)
+    {
+        return client.doesObjectExist(bucketName, path);
+    }
+
+    public void createFolder(String path)
+    {
+        // Check to make sure directory does not already exist
+        if (exists(ensureTrailingSlashPresent(path))) {
+            return;
+        }
+
+        List<String> parts = Splitter.on('/').splitToList(path);
+        StringBuilder sb = new StringBuilder();
+
+        // Loop through all folders in path and create folders if not already exists
+        for (String objectName : parts) {
+            sb.append(ensureTrailingSlashPresent(objectName));
+            if (!exists(sb.toString())) {
+                createEmptyObject(sb.toString());
+            }
+        }
+    }
+
+    public void deleteFolder(String path)
+    {
+        // Retrieve all objects underneath this folder (including folders)
+        List<String> objects = getAllObjects(ensureTrailingSlashPresent(path));
+
+        // For all objects retrieved, delete
+        for(String objectName : objects) {
+            deleteObject(objectName);
+        }
+
+        // Delete the folder itself
+        deleteObject(ensureTrailingSlashPresent(path));
+    }
+
+    public void saveFile(String path, InputStream stream, long byteLength)
+    {
+        ObjectMetadata metadata = new ObjectMetadata();
+        metadata.setContentLength(byteLength);
+
+        client.putObject(new PutObjectRequest(bucketName, path, stream, metadata));
+
+        logger.debug("Saved file {} to S3", path);
+    }
+
+    public void loadFile(String path, OutputStream outputStream)
+    {
+        try {
+            S3Object object = client.getObject(new GetObjectRequest(bucketName, path));
+            IOUtils.copy(object.getObjectContent(), outputStream);
+        } catch (IOException exception) {
+            throw new RuntimeException("Could not read file " + path + " from S3", exception);
+        }
+
+        logger.debug("Loaded file {} from S3", path);
+    }
+
+    public String loadFile(String path)
+    {
+        ByteArrayOutputStream output = new ByteArrayOutputStream();
+        loadFile(path, output);
+        return new String(output.toByteArray(), defaultCharset());
+    }
+
+    public void deleteFile(String path)
+    {
+        boolean isFolder = path.substring(path.length() - 1).equals("/");
+
+        // If not a folder, delete file; otherwise throw exception
+        if (!isFolder) {
+            deleteObject(path);
+        } else {
+            throw new RuntimeException("Folder was passed in to deleteFile: " + path);
+        }
+    }
+
+    /**
```

Comment:
I do not think grouping public and private methods in blocks is best file organization. Typically the code is easiest to read if method appears  first just below the method which uses it for the first time. You can have exceptions from that for sure but it is good rule of thumb.